src/main/java/io/leonis/algieba/control/anomaly/HBOSAnomalyDetector.java

```diff
@@ -0,0 +1,58 @@
+package io.leonis.algieba.control.anomaly;
+
+import io.leonis.algieba.control.*;
+import io.leonis.algieba.statistic.*;
+import java.util.stream.Collectors;
+import lombok.Value;
+
+/**
+ * The Class HBOSAnomalyDetector.
+ *
+ * This class describes an {@link AnomalyDetector} which uses the <a
+ * href="http://www.dfki.de/KI2012/PosterDemoTrack/ki2012pd13.pdf">HBOS algorithm</a>.
+ *
+ * @author Rimon Oz
+ */
+@Value
+public class HBOSAnomalyDetector implements ScanningAnomalyDetector<Histogram, Number> {
+  private final double weight;
+  private final double maximumScore;
+
+  @Override
+  public boolean test(final Histogram inputSpaceRepresentation, final Number input) {
+    return this.computeHBOS(inputSpaceRepresentation, input).doubleValue() > this.getMaximumScore();
+  }
+
+  /**
+   * Computes the Histogram-Based Outlier Score (HBOS) of the supplied value, according to the
+   * supplied measurement space.
+   *
+   * @param inputSpaceRepresentation The measurement space representation as a {@link Histogram}.
+   * @param value                    The value to compute the HBOS of.
+   * @return The HBOS of the supplied value w.r.t. the input space representation.
+   */
+  protected Number computeHBOS(final Histogram inputSpaceRepresentation, final Number value) {
```

Comment:
What does the `protected` modifier mean in a final class? e.g. if a class can't be subclassed.